prow/plugins/skip/skip.go

```diff
@@ -105,14 +94,17 @@ func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent, p
 	}
 
 	for _, job := range presubmits {
-		// Ignore blocking jobs.
-		if !job.SkipReport && job.AlwaysRun {
-			continue
+		// Ignore jobs that report and should run.
+		shouldRun, err := job.ShouldRun(pr.Base.Ref, e.Body, config.NewGitHubDeferredChangedFilesProvider(gc, org, repo, number))
+		if err != nil {
+			resp := fmt.Sprintf("Cannot get changes for PR #%d in %s/%s: %v", number, org, repo, err)
+			log.Warn(resp)
+			return gc.CreateComment(org, repo, number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, e.User.Login, resp))
 		}
-		// Ignore jobs that need to run against the PR changes.
-		if !job.SkipReport && job.RunIfChanged != "" && job.RunsAgainstChanges(changes) {
+		if job.ContextRequired() && shouldRun {
```

Comment:
This isn't technically the same but in the past didn't check that the job ran against the branch or was explicitly called out in the body. Should be fine.